modules/sievefilters/site.js

```diff
@@ -539,12 +532,6 @@ $(function () {
             $('#stop_filtering').prop('checked', false);
             current_account = $(this).attr('account');
             edit_filter_modal.open();
-
-            // Reset the form fields when opening the modal
-            $(".modal_sieve_filter_name").val('');
```

Comment:
By deleting this code, the form starts to keep the previous data again because it is this code which resets when we click on add again.